src/validators/validators.ts

```diff
@@ -7,16 +7,16 @@
 
 import { IFormContext } from '../components/FormContext';
 import { TBasicFieldValue } from '../components/withField';
-import { FieldErrorMessageId, TAsyncValidator, TFieldError, TValidator } from './validators.types';
+import { FieldErrorMessageId, TAnyValidator, TFieldError } from './validators.types';
 
 /**
  * Wrapper function to call validators with parameters
  * @param validator function to call
  * @param context form context
  * @param args parameters for the validator
  */
-const withParam = <TCallback extends TValidator | TAsyncValidator>(validator: TCallback, ...args: unknown[]): TCallback => {
```

Comment:
@maschino 
I also had to change the return type here to `TAnyValidator`. I am not sure how the union type return values work in combination with extends.
It's not the best solution but it works for now